assets/js/googlesitekit/datastore/site/info.js

```diff
@@ -159,12 +160,29 @@ export const selectors = {
 	 * @since 1.7.0
 	 *
 	 * @param {Object} state Data store's state.
+	 * @param {(string|undefined)} page Optional page query argument to add to admin URL. If not provided, the base admin URL is returned.
+	 * @param {(Object|undefined)} args Optional additional query arguments to add to admin URL.
 	 * @return {(string|undefined)} This site's admin URL.
 	 */
-	getAdminURL: createRegistrySelector( ( select ) => () => {
+	getAdminURL: createRegistrySelector( ( select ) => ( state, page, args = {} ) => {
 		const { adminURL } = select( STORE_NAME ).getSiteInfo() || {};
 
-		return adminURL;
+		// Return adminURL if undefined, or if no page supplied
+		if ( typeof adminURL === 'undefined' || typeof page === 'undefined' ) {
+			return adminURL;
+		}
+
+		// Prevent a page in args from overriding main page argument
+		const { page: extraPage, ...queryArgs } = args; // eslint-disable-line no-unused-vars
```

Comment:
Good point about adding the comment, this will clarify it. The logic doesn't need to be changed, having `page` as the first query parameter makes sense.

Reply:
Yeah, I figured we wanted the page to be first rather than tacked on to the end, hence this (admittedly slightly odd) approach.
@felixarntz @aaemnnosttv Just to clarify, I should leave L176 as is, and update the comment at L175 to: `// Since page should be first query arg, create queryArgs without 'page' to prevent a 'page' in args from overriding it`?

Reply:
@jqlee85 Sounds good to me (just with a period at the end)!

Reply:
Oh, just now seeing the convo below that may affect this. 😆